server/src/main/java/org/apache/druid/client/selector/ServerSelectorStrategy.java

```diff
@@ -33,6 +35,17 @@
 })
 public interface ServerSelectorStrategy
 {
+  default <T> QueryableDruidServer pick(Query<T> query, Set<QueryableDruidServer> servers, DataSegment segment)
```

Comment:
Similarly, I think it would be better that old interfaces have a default implementation that calls new interfaces with `null` query. Also please annotate `query` with `@Nullable`.

Reply:
done